lib/alchemy/upgrader/tasks/picture_gallery_upgrader.rb

```diff
@@ -17,11 +17,27 @@ def convert_picture_galleries
           return
         end
 
+        db_picture_gallery_names_with_parent_names = Alchemy::Element.where("name LIKE '%_picture_gallery'").map do |gallery_element|
+          [
+            gallery_element.parent_element.name,
+            gallery_element.name
+          ]
+        end.uniq
+
         elements_with_picture_gallery, all_other_elements = config.partition do |e|
           e['picture_gallery']
         end
 
-        if elements_with_picture_gallery.empty?
+        config_picture_gallery_names_with_parent_names = elements_with_picture_gallery.map do |e|
+          [
+            e['name'],
+            "#{e['name']}_picture_gallery"
+          ]
+        end
+
+        all_picture_gallery_names_with_parent_names = config_picture_gallery_names_with_parent_names | db_picture_gallery_names_with_parent_names
+
+        if all_picture_gallery_names_with_parent_names.empty?
```

Comment:
You load to sets of elements union them but don't use them below this line. Why load them in the first place?